sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -372,6 +372,59 @@ case class GpuCast(
       case (ShortType | IntegerType | LongType | ByteType | StringType, BinaryType) =>
         input.getBase.asByteList(true)
 
+      case (IntegerType | LongType, dt: DecimalType) =>
+        assert(ansiMode, "GpuCastToDecimal can only run under ansiMode")
+
+        // Check whether max precision of target decimal exceeds DType.DECIMAL64_MAX_PRECISION,
+        // so the sanity range is relied on the scale of target DecimalType.
+        assertValuesInRange(input.getBase,
+          Scalar.fromDouble(-math.pow(10, DType.DECIMAL64_MAX_PRECISION - dt.scale)),
+          Scalar.fromDouble(math.pow(10, DType.DECIMAL64_MAX_PRECISION - dt.scale)))
+
+        if (dt.scale < 0) {
+          // Rounding is essential when scale is negative,
+          // so we apply HALF_UP rounding manually to keep align with CpuCast.
+          withResource(input.getBase.castTo(DType.create(DType.DTypeEnum.DECIMAL64, 0))) { zero =>
+            zero.round(dt.scale, ai.rapids.cudf.RoundMode.HALF_UP)
+          }
+        } else if (dt.scale > 0) {
+          // Integer will be enlarged during casting if scale > 0, so we cast input to INT64
+          // before casting it to decimal in case of overflow.
+          withResource(input.getBase.castTo(DType.INT64)) { long =>
+            long.castTo(DType.create(DType.DTypeEnum.DECIMAL64, -dt.scale))
+          }
+        } else {
+          input.getBase.castTo(DType.create(DType.DTypeEnum.DECIMAL64, -dt.scale))
+        }
+
+      case (FloatType | DoubleType, dt: DecimalType) =>
+        assert(ansiMode, "GpuCastToDecimal can only run under ansiMode")
+
+        // Approach to minimize difference between CPUCast and GPUCast:
+        // step 1. cast input to FLOAT64 (if necessary)
+        // step 2. cast FLOAT64 to container DECIMAL (who keeps one more digit for rounding)
+        // step 3. perform HALF_UP rounding on container DECIMAL
+        // Corner case: If target scale reaches DECIMAL64_MAX_PRECISION, container DECIMAL can not
+        // be created because of precision overflow. In this case, we perform casting op directly.
```

Comment:
We need to document this in the compatibility document so users understand that our cast is not 100% perfect. We also need a follow on issue to figure out how to make it 100% correct.

Reply:
Compatibility document has been updated. And I will create the issue in soon.